webapp/channels/src/components/root/root.tsx

```diff
@@ -573,7 +573,7 @@ export default class Root extends React.PureComponent<Props, State> {
 export function doesRouteBelongToTeamControllerRoutes(pathname: RouteComponentProps['location']['pathname']): boolean {
     // Note: we have specifically added admin_console to the negative lookahead as admin_console can have integrations as subpaths (admin_console/integrations/bot_accounts)
     // and we don't want to treat those as team controller routes.
-    const TEAM_CONTROLLER_PATH_PATTERN = new RegExp(`^\/(?!admin_console)([a-z0-9\-_]+)\/(channels|messages|threads|drafts|integrations|emoji|${SCHEDULED_POST_URL_SUFFIX})(\/.*)?$`);
+    const TEAM_CONTROLLER_PATH_PATTERN = new RegExp(`^/(?!admin_console)([a-z0-9-_]+)/(channels|messages|threads|drafts|integrations|emoji|${SCHEDULED_POST_URL_SUFFIX})(/.*)?$`);
```

Comment:
what changed here

Reply:
The linter was complaining of those unnecessary `\` as we're defining regex in string, those aren't needed. They were needed much earlier was this regex pattern was not a string but directly defined like `/<pattern>/`

Reply:
After having used backticked strings in Go yesterday to write a regex specifically because it doesn't use backslashes for escaping, this took me a moment to wrap my head around 😅

I think we should actually still escape that hyphen due to some quirks of regular expressions. The version you had looks totally correct, but I tend to be extra cautious with escaping those expressions because of how subtly different it can handle hyphens. It's so easy something like `[a-]` which matches either of those characters to become `[a-z]` which matches any lower case Latin letter and not hyphens, so I always escape hyphens in those cases even if it's not technically needed

```suggestion
    const TEAM_CONTROLLER_PATH_PATTERN = new RegExp(`^/(?!admin_console)([a-z0-9\\-_]+)/(channels|messages|threads|drafts|integrations|emoji|${SCHEDULED_POST_URL_SUFFIX})(/.*)?$`);
```

Thank you for reading this episode of Harrison Has Opinions On Regexes

Reply:
Done.